detectors/host/host.go

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package host // import "go.opentelemetry.io/contrib/detectors/host"
+
+import (
+	"context"
+	"net"
+	"os"
+	"runtime"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+)
+
+type config struct {
+	optInIPAddresses  bool
+	optInMACAddresses bool
+}
+
+func newConfig(options ...Option) *config {
+	c := &config{}
+	for _, option := range options {
+		option.apply(c)
+	}
+
+	return c
+}
+
+// Option applies an Azure VM detector configuration option.
+type Option interface {
+	apply(*config)
```

Comment:
An allocation can be avoided here:

```suggestion
	apply(config) config
```

https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md#config